### DISCUSSION TITLE
Jobs stop running after 20-30 minutes!

### DISCUSSION THREAD

**Discussion by fox-15** [quartznet/quartznet#2144] [Q&A]
Hello!
My ASP.NET Core MVC .NET7 project, I am using Quartz to systematically execute "HelloWorldJob" jobs.
The deployed Web application works great, Quartz jobs run as expected
according to the settings "0 0/5 * * * ?" every 5 minutes, but only for 20-30 minutes.
If no one has visited the WEB site and there have been no new sessions for 20-30 minutes, then the “HelloWorldJob” task stop being executed. But if someone starts using a WEB application, then the "HelloWorldJob" tasks resume their work. How application inactivity can affect performance
Quartz assignments? Please tell me how to fix this problem?

ASP.NET Core MVC .NET7
Quartz.AspNetCore version 3.7.0

Example of my code:
......
//Add Quartz services
services.AddSingleton<IJobFactory, ScopedJobFactory>();
services.AddSingleton<ISchedulerFactory, StdSchedulerFactory>();
services.AddHostedService<Config.Quartz.QuartzHostedService>();
services.AddSingleton<QuartzJobRunner>();

//Job
services.AddScoped<HelloWorldJob>();
services.AddSingleton(new JobSchedule(
    jobType: typeof(HelloWorldJob), cronExpression: "0 0/5 * * * ?"));   //Every 5 minutes
......

**Comment by fox-15** [comment]
It seems to me that the inactive process is jamming IIS
Quartz has nothing to do with this

**Reply by jafin** [reply]
This sounds like your IIS AppPool shutting down, recycling..   Some suggestions for keeping it running https://docs.hangfire.io/en/latest/deployment-to-production/making-aspnet-app-always-running.html

**Comment by fox-15** [comment]
Application Pools
Idle timeout (in minutes) = 20 (default)
